### PR TITLE
Feat/redis reset

### PR DESCRIPTION
**Pull request by mikael-bashir**
- modified 2 data.ts functions to now use Redis for password-reset token storage/retrieval
- deleted the old functions, making sure new ones match previous input/output/logs
- added initialisation for Redis in lib folder

### CODE REVIEW

File: app/api/forgotten-password/route.ts
Comment (AnishKochhar) [inline app/api/forgotten-password/route.ts:1]: Just to need to update comment on line 16 with what 'generateResetToken' actually done (redis)

File: app/lib/data.ts [hunk 1]
```diff
@@ -374,6 +375,28 @@ export async function insertUser(formData: UserRegisterFormData) {
...
+	try {
+		const hashedPassword = await bcrypt.hash(password, 10);
+		await sql`
+			UPDATE users
+			SET 
+				password = ${hashedPassword}
+			WHERE email = ${email} --- Email is UNIQUE among users table
+		`
+		// Delete the reset_password entry
+		await sql`
```
Comment (AnishKochhar) [inline app/lib/data.ts:389]: Need to remove this sql DELETE

File: changelog.md [hunk 2]
```diff
@@ -4,6 +4,31 @@ All the notable additions and fixes.
...
 
+# [1.1.5] - 2024-12-27
+
+### Changed
+
+- Updated login form to use react-hook-form
+
+### Fixed
+
+- compile error due to width and height not specified for Image component in account page
```
Comment (AnishKochhar) [inline changelog.md:15]: Capitalise please

**Review by AnishKochhar — COMMENTED** [review]
Lovely code. Some very minor formatting, and removing of redundant code

**Review by AnishKochhar — APPROVED** [review]

**Review by mikael-bashir — COMMENTED** [review]
Approved!

**Review by AnishKochhar — APPROVED** [review]

**Pull request merged**